src/app/presence-control/services/presence-control-detail-service.ts

```diff
@@ -143,17 +149,30 @@ export class PresenceControlDetailService {
   ): ApprenticeshipCompany {
     const apprenticeshipCompany: ApprenticeshipCompany = {
       apprenticeshipContract: contract,
-      jobTrainerPerson: contract.JobTrainer
-        ? this.findPerson(contract.JobTrainer, persons)
+      jobTrainer: contract.JobTrainer
+        ? this.mapToProfilePerson(this.findPerson(contract.JobTrainer, persons))
         : null,
-      apprenticeshipManagerPerson: this.findPerson(
-        contract.ApprenticeshipManagerId,
-        persons
+      apprenticeshipManager: this.mapToProfilePerson(
+        this.findPerson(contract.ApprenticeshipManagerId, persons)
       ),
     };
     return apprenticeshipCompany;
   }
 
+  private mapToProfilePerson(person: Person): ProfilePerson {
+    return {
+      person,
+      email: this.getEmail(person),
+    };
+  }
+
+  private getEmail(person: Person): Option<string> {
+    const email = [person.DisplayEmail, person.Email, person.Email2].find(
+      (e) => e
+    );
+    return email ? email : null;
+  }
+
```

Comment:
Du hast jetzt recht viel anpassen müssen, wegen der `ProfilePerson` die du hier im Service erstellst. Da es vorallem um eine Darstellungssache geht, hätte ich dies bei der Anzeige (d.h. in der Component) angepasst. Es würde also nur an drei Orten im Template eine Anpassung brauchen. Möglicherweise könntest du dies sogar mit einer eigenen Pipe lösen, z.B.:
```
<a href="mailto:{{ apprenticeshipCompany.apprenticeshipManagerPerson | erzPersonEmail }}">
  {{ apprenticeshipCompany.apprenticeshipManagerPerson | erzPersonEmail }}
</a>
```

Für die Implementation brauchst du auch kein `.find()` oder so (das `|| null` wäre nur nötig, wenn `Email2` auch `undefined` sein könnte):
```
@Pipe({
  name: 'erzPersonEmail'
})
export class PersonEmailPipe implements PipeTransform {
  transform(input: Person): Option<string> {
    return person.DisplayEmail || person.Email || person.Email2 || null;
  }
}
```

Von der Performance her wäre das sicher auch kein Problem, wenn du das an mehreren Orten im Template brauchst, ist ja eine einfache `||` Operation.

Das ist alles natürlich nicht getestet™

Reply:
Ich habe das entsprechend umgebaut (unter der Annahme dass die Emails nie `undefined` sind).